testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/single/web/shared/SessionDestroyCounter.java

```diff
@@ -0,0 +1,17 @@
+package org.jboss.as.test.clustering.single.web.shared;
+
+import jakarta.ejb.Singleton;
+
+@Singleton
+public class SessionDestroyCounter {
+
+    private int sessionDestroyCount = 0;
+
+    public int getSessionDestroyCount() {
+        return this.sessionDestroyCount;
+    }
+
+    public void incrementSessionDestroyCount() {
+        this.sessionDestroyCount++;
+    }
```

Comment:
This is vulnerable to memory visibility issues, since incrementSessionDestroyCount() and getSessionDestroyCount() will be invoked by different threads. I would suggest using AtomicInteger to additionally ensure atomicity of your increment operation.